File: UrbanDictionary/app/src/main/java/com/nike/urbandictionary/adapters/UrbanDictionaryAdapter.java
package com.nike.urbandictionary.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.nike.urbandictionary.R;
import com.nike.urbandictionary.models.UrbanDictionaryDefinition;

import java.util.List;

public class UrbanDictionaryAdapter extends RecyclerView.Adapter<UrbanDictionaryAdapter.UrbanDictionaryViewHolder> {

    private Context context;
    private List<UrbanDictionaryDefinition> definitions = null;

    public UrbanDictionaryAdapter(Context context, List<UrbanDictionaryDefinition> definitions) {
        this.context = context;
        this.definitions = definitions;
    }

    @NonNull
    @Override
    public UrbanDictionaryAdapter.UrbanDictionaryViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.definition_item, parent, false);
        return new UrbanDictionaryViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull UrbanDictionaryAdapter.UrbanDictionaryViewHolder holder, int position) {
        holder.definition.setText(definitions.get(position).getWordDefinition());
        holder.thumbsup.setText(definitions.get(position).getThumbsUpCount());
        holder.thumbsdown.setText(definitions.get(position).getThumbsDownCount());
    }

    @Override
    public int getItemCount() {
        return definitions.size();
    }

    class UrbanDictionaryViewHolder extends RecyclerView.ViewHolder {

        TextView definition;
        TextView thumbsup;
        TextView thumbsdown;

        UrbanDictionaryViewHolder(@NonNull View itemView) {
            super(itemView);
            definition = itemView.findViewById(R.id.definition);
            thumbsup = itemView.findViewById(R.id.thumbsUp);
            thumbsdown = itemView.findViewById(R.id.thumbsDown);

        }
    }
}

File: UrbanDictionaryKotlin/app/src/main/java/com/nike/urbandictionary/adapters/UrbanDictionaryAdapter.kt
package com.nike.urbandictionary.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import com.nike.urbandictionary.R
import com.nike.urbandictionary.models.UrbanDictionaryDefinition

class UrbanDictionaryAdapter(private val context: Context, private var definitions: List<UrbanDictionaryDefinition>) : RecyclerView.Adapter<UrbanDictionaryAdapter.UrbanDictionaryViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UrbanDictionaryViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.definition_item, parent, false)
        return UrbanDictionaryViewHolder(view)
    }

    override fun onBindViewHolder(holder: UrbanDictionaryViewHolder, position: Int) {
        holder.definition.text = definitions[position].wordDefinition
        holder.thumbsup.text = definitions[position].thumbsUpCount
        holder.thumbsdown.text = definitions[position].thumbsDownCount
    }

    override fun getItemCount(): Int {
        return definitions.size
    }

    class UrbanDictionaryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var definition: TextView = itemView.findViewById(R.id.definition)
        var thumbsup: TextView = itemView.findViewById(R.id.thumbsUp)
        var thumbsdown: TextView = itemView.findViewById(R.id.thumbsDown)
    }
}
